commit 45aab3adfa718688e2b0dd08d948565e5210eaab (refs/changes/43/5343/1)
Author: Rui Loura <rui@joyent.com>
Date:   2019-01-10T17:31:38+00:00 (9 months ago)
    
    TRITON-877 admin NIC functions in net-agent need to be factored out

diff --git a/lib/server-fsm.js b/lib/server-fsm.js
index 4214ec8..2597af9 100644
--- a/lib/server-fsm.js
+++ b/lib/server-fsm.js
@@ -158,7 +158,7 @@ ServerFSM.prototype._update = function (sysinfo) {
             self.nictags[tag] = name;
         });
 
-        watchNic(mac, self._fmtpnic(pnic));
+        watchNic(mac, self._fmtpnic(pnic, sysinfo));
     });
 
     mod_jsprim.forEachKey(vnics, function (name, vnic) {
@@ -186,7 +186,8 @@ ServerFSM.prototype._fmtstate = function (state) {
     return (state === 'up' ? 'running' : 'stopped');
 };
 
-ServerFSM.prototype._fmtpnic = function (nic) {
+ServerFSM.prototype._fmtpnic = function (nic, sysinfo) {
+    var admin_tag = sysinfo['Admin NIC Tag'] || 'admin';
     var o = {
         belongs_to_uuid: this.uuid,
         belongs_to_type: 'server',
@@ -199,16 +200,17 @@ ServerFSM.prototype._fmtpnic = function (nic) {
         o.ip = nic.ip4addr;
     }
 
+    /*
+     * Alternatively we could do 
+     *      var admin_nic = netconfig.adminNicFromSysinfo(sysinfo);
+     *      if (mod_jsprim.deepEqual(admin_nic, nic)) {
+     *          o.nic_tag = sysinfo['Admin NIC Tag'] || 'admin';
+     *          o.vlan_id = 0;
+     *      }
+     */
     /* If this smells like an admin NIC, try to set "nic_tag" */
-    var atags = nic['NIC Names'].filter(isAdminLike);
-    if (atags.length > 0) {
-        if (atags.length > 1) {
-            this.log.warn({
-                nic: nic, matching: atags
-            }, 'multiple candidate tags found for admin-like interface');
-        }
-
-        o.nic_tag = atags[0];
+    if (nic['NIC Names'].indexOf(admin_tag) !== -1) {
+        o.nic_tag = admin_tag;
         o.vlan_id = 0;
     }
 
